File: internal/controllers/ai_controller.go
package controllers

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/growthOS/qr-review-service/internal/dto"
	"github.com/growthOS/qr-review-service/internal/services"
)

// AIController handles HTTP requests for AI review suggestion generation.
type AIController struct {
	service *services.AISuggestionService
}

// NewAIController creates a new AI controller.
func NewAIController(service *services.AISuggestionService) *AIController {
	return &AIController{service: service}
}

// GetSuggestions handles POST /api/v1/qr-reviews/ai/review-suggestions
func (ctrl *AIController) GetSuggestions(c *gin.Context) {
	var req dto.AISuggestionRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{
			"success": false,
			"message": "Invalid request: " + err.Error(),
		})
		return
	}

	// Only generate suggestions for positive ratings
	if req.Rating < 4 {
		c.JSON(http.StatusBadRequest, gin.H{
			"success": false,
			"message": "Review suggestions are only available for ratings of 4 or higher",
		})
		return
	}

	suggestions, err := ctrl.service.GenerateSuggestions(req.BusinessType, req.City, req.Rating)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"success": false,
			"message": "Failed to generate suggestions: " + err.Error(),
		})
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"success": true,
		"data":    suggestions,
	})
}

File: backend/src/controllers/aiController.js
const aiSuggestionService = require("../services/aiSuggestionService");

async function getSuggestions(req, res) {
  const { business_type, city, rating } = req.body;
  if (!business_type || !city || !rating || rating < 1 || rating > 5) {
    return res.status(400).json({
      success: false,
      message:
        "Invalid request: business_type, city, and rating (1-5) are required",
    });
  }

  if (rating < 4) {
    return res.status(400).json({
      success: false,
      message: "Review suggestions are only available for ratings of 4 or higher",
    });
  }

  try {
    const suggestions = await aiSuggestionService.generateSuggestions(
      req.body.business_name,
      business_type,
      city,
      rating,
      req.body.service_taken,
      req.body.language
    );
    res.status(200).json({ success: true, data: suggestions });
  } catch (err) {
    res.status(500).json({
      success: false,
      message: "Failed to generate suggestions: " + err.message,
    });
  }
}

module.exports = { getSuggestions };
